src/adapter/src/catalog.rs

```diff
@@ -1631,9 +1641,26 @@ impl ConnCatalog<'_> {
         &*self.state
     }
 
+    /// Prevent planning from resolving item with the provided ID. Instead,
+    /// return an error as if the item did not exist.
+    ///
+    /// This feature is meant exclusively to permit re-planning statements
+    /// during update operations.
+    ///
+    /// # Panics
+    /// If the catalog's role ID is not [`MZ_SYSTEM_ROLE_ID`].
+    pub fn mark_id_unresolvable(&mut self, id: GlobalId) {
+        assert_eq!(
+            self.role_id, MZ_SYSTEM_ROLE_ID,
+            "only the system role can mark IDs unresolvable",
```

Comment:
Seems like we never have a failure for this assertion.

Reply:
Just for my own understanding going forward, what is the expectation here?

Reply:
This is a panic, right? In this case I guess it's fine, no need to do anything, I shouldn't have marked it.

Reply:
A panic, yes––and nbd just wanted to make sure I understood :)